test/framework/management/kind/mgmt.go

```diff
@@ -49,33 +55,25 @@ type Cluster struct {
 
 // NewCluster sets up a new kind cluster to be used as the management cluster.
 func NewCluster(ctx context.Context, name string, scheme *runtime.Scheme, images ...string) (*Cluster, error) {
-	cmd := exec.NewCommand(
-		exec.WithCommand("kind"),
-		exec.WithArgs("create", "cluster", "--name", name),
-	)
-	return create(ctx, cmd, name, scheme, images...)
+	p := cluster.NewProvider(cluster.ProviderWithLogger(cmd.NewLogger()))
+	return create(ctx, p, name, nil, scheme, images...)
 }
 
 // NewClusterWithConfig creates a kind cluster using a kind-config file.
 func NewClusterWithConfig(ctx context.Context, name, configFile string, scheme *runtime.Scheme, images ...string) (*Cluster, error) {
-	cmd := exec.NewCommand(
-		exec.WithCommand("kind"),
-		exec.WithArgs("create", "cluster", "--name", name, "--config", configFile),
-	)
-	return create(ctx, cmd, name, scheme, images...)
+	p := cluster.NewProvider(cluster.ProviderWithLogger(cmd.NewLogger()))
+	config := cluster.CreateWithConfigFile(configFile)
+	return create(ctx, p, name, config, scheme, images...)
 }
 
-func create(ctx context.Context, cmd *exec.Command, name string, scheme *runtime.Scheme, images ...string) (*Cluster, error) {
+func create(ctx context.Context, provider *cluster.Provider, name string, config cluster.CreateOption, scheme *runtime.Scheme, images ...string) (*Cluster, error) {
 	f, err := ioutil.TempFile("", "mgmt-kubeconfig")
 	// if there is an error there will not be a file to clean up
 	if err != nil {
 		return nil, err
 	}
 	// After this point we have things to clean up, so always return a *Cluster
 
-	// Write the kubeconfig directly to the temp file
-	cmd.Args = append(cmd.Args, "--kubeconfig", f.Name())
```

Comment:
Without overriding the Kubeconfig, Kind will be reading and writing the default Kubeconfig file `~/.kube/config`, this probably isn't desirable during test runs is it?
Might it be possible to retain the existing behaviour somehow?
Looks like it might be possible using: https://github.com/kubernetes-sigs/kind/blob/93abedaa23f6d4639e0c1b83fc0eb7cacbf357e4/pkg/cluster/createoption.go#L101-L107

Reply:
Totally agree here

Reply:
This is resolved 👍